packages/blade/src/components/Amount/amountTokens.ts

```diff
@@ -1,48 +1,102 @@
-// @ts-nocheck
-import type { AmountProps } from './Amount';
 import type { FontSize, Typography } from '~tokens/global';
+import type { BaseTextProps } from '~components/Typography/BaseText/types';
 
-const affixFontSizes: Record<NonNullable<AmountProps['size']>, keyof FontSize> = {
-  'body-small': 75,
-  'body-small-bold': 75,
-  'body-medium': 75,
-  'body-medium-bold': 75,
-  'heading-small': 75,
-  'heading-small-bold': 75,
-  'heading-large': 100,
-  'heading-large-bold': 100,
-  'title-small': 300,
-  'title-medium': 400,
-} as const;
+type AmountSizes = 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge' | '2xlarge';
 
-const amountFontSizes: Record<NonNullable<AmountProps['size']>, keyof FontSize> = {
-  'body-small': 75,
-  'body-small-bold': 75,
-  'body-medium': 100,
-  'body-medium-bold': 100,
-  'heading-small': 200,
-  'heading-small-bold': 200,
-  'heading-large': 400,
-  'heading-large-bold': 400,
-  'title-small': 600,
-  'title-medium': 700,
-} as const;
+type AmountDisplayProps = {
+  type?: 'display';
+  size?: Extract<AmountSizes, 'small' | 'medium' | 'large' | 'xlarge'>;
+  weight?: Extract<BaseTextProps['fontWeight'], 'regular' | 'medium' | 'semibold'>;
+};
+
+type AmountHeadingProps = {
+  type?: 'heading';
+  size?: Extract<AmountSizes, 'small' | 'medium' | 'large' | 'xlarge' | '2xlarge'>;
+  weight?: Extract<BaseTextProps['fontWeight'], 'regular' | 'semibold'>;
+};
+
+type AmountBodyProps = {
+  type?: 'body';
+  size?: Extract<AmountSizes, 'xsmall' | 'small' | 'medium' | 'large'>;
+  weight?: Extract<BaseTextProps['fontWeight'], 'regular' | 'medium' | 'semibold'>;
+};
+
+type AmountTypeProps = AmountDisplayProps | AmountHeadingProps | AmountBodyProps;
+
+const normalAmountSizes: Record<
```

Comment:
we just need to add run time error for type values that don't have corresponding size values